### PR TITLE
feat: add support for additional dev console fields on create

### PR DESCRIPTION
**Pull request by MichaelGoberling**
## Description

Add support for setting the following fields on template create: 
- author
- adobeRecommended
- status
- codeSamples 

## How Has This Been Tested?

`npm run test`, locally deployed changes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: actions/templates/post/index.js [hunk 1]
```diff
@@ -20,6 +20,28 @@ const consoleLib = require('@adobe/aio-lib-console');
...
+    name: params.name,
+    ...(params.description && { description: params.description }), // developer console only
+    ...(params.version && { version: params.version }), // developer console only
+    ...(params.createdBy && { createdBy: params.createdBy }),
+    ...(params.author && { author: params.author }), // developer console only
+    ...(params.status && { status: params.status }), // developer console only
+    ...(params.adobeRecommended && { adobeRecommended: params.adobeRecommended }), // developer console only
+    ...(params.codeSamples && { codeSamples: params.codeSamples }), // developer console only
+    links: {
+      ...(params?.links?.consoleProject && { consoleProject: params.links.consoleProject }), // developer console only
```
Comment (shazron) [inline actions/templates/post/index.js:39]: I saw in some test data in another PR, `links` can contain an `npm` property also?

File: test/templates.post.test.js [hunk 2]
```diff
@@ -496,6 +496,223 @@ describe('POST templates', () => {
...
+      links: {
+        consoleProject: DEVELOPER_CONSOLE_PROJECT
+      },
+      description: 'Developer Console template',
+      version: '1.0.0',
+      createdBy: 'Capernicus',
+      author: 'Capernicus'
+    });
+    // TODO: Uncomment the following after integrating with App Builder templates again
+    // expect(createReviewIssue).toHaveBeenCalledWith(TEMPLATE_NAME, TEMPLATE_GITHUB_REPO, process.env.ACCESS_TOKEN_GITHUB, process.env.TEMPLATE_REGISTRY_ORG, process.env.TEMPLATE_REGISTRY_REPOSITORY);
```
Comment (shazron) [inline test/templates.post.test.js:712]: perhaps we should add a story for this so we don't forget after

**Review by shazron — APPROVED** [review]

**Review by arjuncooliitr — APPROVED** [review]

File: actions/templates/post/index.js (hunk 1 above)
Comment (MichaelGoberling) [reply actions/templates/post/index.js:39]: Ah yes `links` can have `npm`, however it's not set via request body, but inferred from the template name when inserting a new template [here](https://github.com/adobe/adp-template-registry-api/blob/main/actions/templateRegistry.js#L55)

File: test/templates.post.test.js (hunk 2 above)
Comment (MichaelGoberling) [reply test/templates.post.test.js:712]: Good thinking. Just added one to the backlog

**Pull request merged**